packages/react-native/React/CoreModules/RCTAppearance.mm

```diff
@@ -36,6 +36,18 @@ void RCTOverrideAppearancePreference(NSString *const colorSchemeOverride)
   return sColorSchemeOverride;
 }
 
+UITraitCollection* getKeyWindowTraitCollection() {
+  if ([NSThread isMainThread]) {
+    return [UIApplication sharedApplication].delegate.window.traitCollection;
+  } else {
+    __block UITraitCollection* traitCollection = nil;
+    dispatch_sync(dispatch_get_main_queue(), ^{
+      traitCollection = [UIApplication sharedApplication].delegate.window.traitCollection;
```

Comment:
Please, use the [`RCTSharedApplication`](https://github.com/facebook/react-native/blob/9841bd81852d59608fe3566b17831d6d42eb7dcf/packages/react-native/React/Base/RCTUtils.m#L537)  utility function here